workspaces/announcements/plugins/announcements-backend/src/router.ts

```diff
@@ -384,71 +496,114 @@ export async function createRouter(
         });
       }
 
+      await auditorEvent.success();
       return res.status(204).end();
     },
   );
 
   router.get('/tags', async (_req, res) => {
-    const results = await persistenceContext.tagsStore.tags();
-    return res.json(results);
+    const auditorEvent = await auditor.createEvent({
+      eventId: AUDITOR_FETCH_EVENT_ID,
+      request: _req,
+      severityLevel: 'low',
+      meta: {
+        queryType: 'all',
+      },
+    });
+    try {
+      const results = await persistenceContext.tagsStore.tags();
+      await auditorEvent.success();
+      return res.json(results);
+    } catch (err) {
+      await auditorEvent.fail({ error: err });
+      throw err;
+    }
   });
 
-  router.post(
-    '/tags',
-    async (req: Request<{}, {}, { title: string }, {}>, res) => {
-      if (!(await isRequestAuthorized(req, announcementCreatePermission))) {
-        throw new NotAllowedError('Unauthorized');
-      }
-
-      const { title } = req.body;
-
-      if (!title || typeof title !== 'string' || title.trim() === '') {
-        return res.status(400).json({ error: 'Title is required' });
-      }
-
-      if (title.length > MAX_TITLE_TAG_LENGTH) {
-        return res.status(400).json({ error: 'Title exceeds maximum length' });
-      }
-
-      const slug = slugify(title, { lower: true });
-
-      const existingTag = await persistenceContext.tagsStore.tagBySlug(slug);
-      if (existingTag) {
-        return res.status(409).json({ error: 'Tag already exists' });
-      }
-
-      const tag = {
-        title,
-        slug,
-      };
-
-      await persistenceContext.tagsStore.insert(tag);
-
-      if (events) {
-        events.publish({
-          topic: EVENTS_TOPIC_ANNOUNCEMENTS,
-          eventPayload: {
-            tag: tag.slug,
-          },
-          metadata: { action: EVENTS_ACTION_CREATE_TAG },
-        });
-      }
+  router.post('/tags', async (req: Request<{}, {}, TagsRequest, {}>, res) => {
+    const auditorEvent = await auditor.createEvent({
+      eventId: AUDITOR_MUTATE_EVENT_ID,
+      request: req,
+      severityLevel: 'medium',
+      meta: {
+        actionType: AUDITOR_ACTION_UPDATE,
+      },
+    });
+
+    if (!(await isRequestAuthorized(req, announcementCreatePermission))) {
+      const error = new NotAllowedError('Unauthorized');
+      await auditorEvent.fail({ error });
+      throw error;
+    }
+
+    const title = req.body.title;
+
+    if (!title || typeof title !== 'string' || title.trim() === '') {
```

Comment:
A nice follow up would be to leverage zod where possible to do handle parsing validation like this. No need to change here because I see we are doing it in multiple places.